### PR TITLE
Safe ccall and objects

### PR DESCRIPTION
**Pull request by musm**
use `unsafe_convert`

### CODE REVIEW

File: src/libmagickwand.jl [hunk 1]
```diff
@@ -138,37 +138,65 @@ const NoCompression = 1
 
 type MagickWand
     ptr::Ptr{Void}
+
+    function MagickWand()
+        ptr = ccall((:NewMagickWand, libwand), Ptr{Void}, ())
+        obj = new(ptr)
```
Comment (musm) [inline src/libmagickwand.jl]: should probably add `ptr == C_NULL && throw(OutOfMemoryError())`    and same below

**Comment by timholy** [comment]
I like it. OK to merge?

**Comment by musm** [comment]
Yep!

**Comment by timholy** [comment]
Thanks a bunch, @musm.

**Pull request merged**